dev-util/gammaray/gammaray-2.9.0.ebuild

```diff
@@ -0,0 +1,90 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+QT_MINIMAL="5.9"
+
+inherit cmake-utils kde5-functions
+
+DESCRIPTION="High-level runtime introspection tool for Qt applications"
+HOMEPAGE="https://github.com/KDAB/GammaRay"
+
+if [[ ${PV} == 9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/KDAB/GammaRay.git"
+else
+	SRC_URI="https://github.com/KDAB/GammaRay/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="BSD"
+SLOT=0
+IUSE="auto-detect backtrace bluetooth +client\
+	concurrent designer -doc geolocation\
+	printsupport scxml svg test qt3d qml unwind\
+	wayland webengine"
+
+RDEPEND="
+	$(add_qt_dep qtcore)
+	$(add_qt_dep qtgui)
+	$(add_qt_dep qtnetwork)
+	$(add_qt_dep qtwidgets)
+	$(add_qt_dep qtxml)
+	$(add_frameworks_dep kitemmodels)
+	qt3d? ( $(add_qt_dep qt3d) )
+	bluetooth? ( $(add_qt_dep qtbluetooth) )
+	concurrent? ( $(add_qt_dep qtconcurrent) )
+	designer? ( $(add_qt_dep designer) )
+	geolocation? ( $(add_qt_dep qtpositioning) )
+	printsupport? ( $(add_qt_dep qtprintsupport) )
+	qml? ( $(add_qt_dep qtdeclarative 'widgets') )
+	scxml? ( $(add_qt_dep qtscxml) )
+	svg? ( $(add_qt_dep qtsvg) )
+	webengine? ( $(add_qt_dep qtwebengine 'widgets') )
+	wayland? ( $(add_qt_dep qtwayland) )
+"
+
+DEPEND="${RDEPEND}
+	test? ( $(add_qt_dep qttest) )
+"
+
+src_configure(){
+	local mycmakeargs+=(
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dAnimation=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dExtras=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dInput=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dLogic=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dRender=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5qt3dQuick=$(usex !qt3d)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Bluetooth=$(usex !bluetooth)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Concurrent=$(usex !concurrent)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Designer=$(usex !designer)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Positioning=$(usex !geolocation)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5PrintSupport=$(usex !printsupport)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Qml=$(usex !qml)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Quick=$(usex !qml)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5QuickWidgets=$(usex !qml)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Svg=$(usex !svg)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Scxml=$(usex !scxml)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Test=$(usex !test)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5WebEngineWidgets=$(usex !webengine)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5Widgets=$(usex !webengine)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Qt5WaylandCompositor=$(usex !wayland)
+		-DGAMMARAY_BUILD_DOCS=$(usex doc)
+		-DGAMMARAY_BUILD_UI=$(usex client)
+		-DGAMMARAY_DISABLE_FEEDBACK=ON
+		-DSTACK_DETAILS_AUTO_DETECT=$(usex auto-detect)
```

Comment:
Those flags still need figuring out. Did you check what they do? Can you check upstream recommendations somewhere?
```
set(STACK_DETAILS_AUTO_DETECT TRUE CACHE BOOL
        "Auto detect backward's stack details dependencies")
```
^ we probably don't want that at all (no build-time automagic).

Please have a look inside `3rdparty/backward-cpp/BackwardMacros.cmake`, the hierarchy of `STACK_DETAILS_AUTO_DETECT` block may give you some clues about upstream's preferences. And you will see that choices are made based on the availability of further dependencies like `dev-libs/elfutils`, `sys-libs/binutils-libs`.

`STACK_WALKING_UNWIND` and `STACK_WALKING_BACKTRACE` both do not seem to do anything, they are just sitting in that file being defined as options that are never checked.